modules/storage/main.tf

```diff
@@ -0,0 +1,7 @@
+module "portworx" {
+  count            = var.storage_module == "portworx" ? 1 : 0
+  source           = "../portworx"
+  portworx_version = try(var.storage_options.version, null)
```

Comment:
`storage_options` are arbitrary.

The `storage` module could insist that options are prefixed by their module name.
If we do this, it will be easier to keep portworx settings next to the settings of other modules, so only the module name needs to be changed in `terraform.tfvars`:

```hcl
storage_module = "portworx"
// storage_module = "rook"
storage_options = {
  portworx_version = "2.6"
  rook_version = "1.5.6"
}
```

vs

```hcl
storage_module = "portworx"
storage_options = {
  version = "2.6"
}

// storage_module = "rook"
// storage_options = {
//  version = "1.5.6"
// }
```

In practice, this shouldn't be too confusing or make much of a difference.

Reply:
Names without vendor prefixes could be used to allow arguments that are shared between vendors.

Reply:
I added vendor prefixes to the variables.